crates/sncast/src/helpers/braavos.rs

```diff
@@ -0,0 +1,126 @@
+use async_trait::async_trait;
+use starknet::{
+    accounts::{AccountFactory, PreparedAccountDeployment, RawAccountDeployment},
+    core::types::{BlockId, BlockTag, FieldElement},
+    providers::Provider,
+    signers::Signer,
+};
+use starknet_crypto::poseidon_hash_many;
+
+// Adapted from strakli as there is currently no implementation of braavos account factory in starknet-rs
+pub struct BraavosAccountFactory<S, P> {
+    class_hash: FieldElement,
+    base_class_hash: FieldElement,
+    chain_id: FieldElement,
+    signer_public_key: FieldElement,
+    signer: S,
+    provider: P,
+    block_id: BlockId,
+}
+
+impl<S, P> BraavosAccountFactory<S, P>
+where
+    S: Signer,
+{
+    pub async fn new(
+        class_hash: FieldElement,
+        base_class_hash: FieldElement,
+        chain_id: FieldElement,
+        signer: S,
+        provider: P,
+    ) -> Result<Self, S::GetPublicKeyError> {
+        let signer_public_key = signer.get_public_key().await?;
+        Ok(Self {
+            class_hash,
+            base_class_hash,
+            chain_id,
+            signer_public_key: signer_public_key.scalar(),
+            signer,
+            provider,
+            block_id: BlockId::Tag(BlockTag::Latest),
+        })
+    }
+
+    pub fn set_block_id(&mut self, block_id: BlockId) -> &Self {
+        self.block_id = block_id;
+        self
+    }
+}
+
+#[cfg_attr(not(target_arch = "wasm32"), async_trait)]
+#[cfg_attr(target_arch = "wasm32", async_trait(?Send))]
```

Comment:
```suggestion
#[async_trait]
```

we don't use wasm anyway